packages/abi-utils/lib/arbitrary.ts

```diff
@@ -0,0 +1,315 @@
+import * as fc from "fast-check";
+import faker from "faker";
+import { camelCase, pascalCase } from "change-case";
+
+import * as Types from "./types";
+
+export const Parameter = () =>
+  fc
+    .tuple(
+      fc.record({
+        name: ParameterName()
+      }),
+      TypeRecord()
+    )
+    .map(records => records.reduce((a, b) => ({ ...a, ...b }), {}));
+
+export const EventParameter = () =>
+  fc
+    .tuple(
+      fc.record({
+        name: ParameterName(),
+        indexed: fc.boolean()
+      }),
+      TypeRecord()
+    )
+    .map(records => records.reduce((a, b) => ({ ...a, ...b }), {}));
+
+export const EventEntry = () =>
+  fc.record({
+    type: fc.constant("event"),
+    name: EventName(),
+    inputs: fc.array(EventParameter(), { maxLength: 10 }).filter(inputs => {
+      if (inputs.filter(({ indexed }) => indexed).length > 3) {
+        // only up to 3 params can be indexed
+        return false;
+      }
+
+      // names that are not blank should be unique
+      const names = inputs.map(({ name }) => name).filter(name => name !== "");
+      return names.length === new Set(names).size;
+    }),
+    anonymous: fc.boolean()
+  });
+
+export const FunctionEntry = () =>
+  fc
+    .tuple(
+      fc.record(
+        {
+          type: fc.constant("function")
+        },
+        { withDeletedKeys: true }
+      ),
+      fc.record({
+        name: FunctionName(),
+        inputs: fc.array(Parameter(), { maxLength: 10 }).filter(inputs => {
+          // names that are not blank should be unique
+          const names = inputs
+            .map(({ name }) => name)
+            .filter(name => name !== "");
+          return names.length === new Set(names).size;
+        })
+      }),
+      fc.record(
+        {
+          outputs: fc.array(Parameter(), { maxLength: 10 }).filter(outputs => {
+            // names that are not blank should be unique
+            const names = outputs
+              .map(({ name }) => name)
+              .filter(name => name !== "");
+            return names.length === new Set(names).size;
+          })
+        },
+        { withDeletedKeys: true }
+      ),
+      fc
+        .tuple(
+          fc.oneof(
+            fc.constant("pure"),
+            fc.constant("view"),
+            fc.constant("nonpayable"),
+            fc.constant("payable")
+          ),
+          fc.boolean()
+        )
+        .map(([stateMutability, includeOptionals]) =>
+          !includeOptionals
+            ? { stateMutability }
+            : {
```

Comment:
I notice you don't include any way of generating ABIs with `stateMutability` missing (in favor of just `payable`/`constant`, as best I can tell.  (This applies to fallbacks and constructors as well, obviously.)  Should that be added, for testing normalization?